code/modules/network/commands/directory_commands.dm

```diff
@@ -0,0 +1,332 @@
+	..()
+	if(!N.root || (N.root.stealth && !ishacktool(H)))
+		feedback = "<span class'notice'>No Root Network.</span>"
+		return feed(H)
+	if(!N.root.password || N.root.password == N.password || N.root.password == A.args[1])
+		connect(N.root, H)
+		return feed(H)
+	if(N.root.password && ishacktool(H) && A.args[1] == "b")
+		feedback = "<span class='warning'>Performing bruteforce hack on [N.root.id].</span>"
+		N.root.execute("security -brute -[N.id]", H) // Trigger the root's security. Loudly.
+		connect(N.root, H)
+		return feed(H)
+	feedback = "<span class='warning'>Encryption Key Mismatch.</span>"
+	N.root.execute("security -alert -[N.id]", H) // Trigger the root's security. Quietly.
+	return feed(H)
+
+
+/datum/network_command/subnet
+	trigger = "subnet"
+	info = "Usage: \"subnet -{target}\". Attempts to connect to the target subnetwork of this network if it exists. If there is only one subnetwork, the target is implicit."
+
+/datum/network_command/subnet/execute(datum/network/N, datum/network_argument/A, obj/item/device/hacktool/H)
+	..()
+	var/datum/network/SN
+	if(A.args[1] && N.sub.len >= 1)
+		for(var/datum/network/S in N.sub)
+			if(S.id == A.args[1])
+				connect(S, H)
+				return feed(H)
+	if(N.sub.len > 1 && !A.args[1])
+		var/not_stealthed = 0
+		for(var/datum/network/S in N.sub)
+			if(!S.stealth)
+				not_stealthed++
+				SN = S
+		if(!ishacktool(H) && not_stealthed == 0)
+			feedback = "<span class='notice'>No subnetworks available. Unable to complete request.</span>"
+			return feed(H)
+		if(ishacktool(H) || not_stealthed > 1)
+			feedback = "<span class='notice'>Multiple subnetworks available. Unable to complete request.</span>"
+			return feed(H)
+		if(!ishacktool(H) && not_stealthed == 1)
+			connect(SN, H)
+			return feed(H)
+	if(N.sub.len == 1)
+		for(var/datum/network/S in N.sub)
+			SN = S
+		if(!SN.stealth || ishacktool(H))
+			connect(SN, H)
+			return feed(H)
+	feedback = "<span class='notice'>No subnetworks available. Unable to complete request.</span>"
+	return feed(H)
+
+
+/datum/network_command/connect
+	trigger = "connect"
+	info = "Usage: \"connect -{target_id} -{encryption_key}\". Will attempt to connect to the target network."
+
+/datum/network_command/connect/execute(datum/network/N, datum/network_argument/A, obj/item/device/hacktool/H)
+	..()
+	if(N.root.id == A.args[1])
+		if(!N.root.password || N.root.password == N.password || N.root.password == A.args[2])
+			connect(N.root, H)
+			return feed(H)
+		if(N.root.password && ishacktool(H) && A.args[2] == "b")
+			feedback = "<span class='warning'>Performing bruteforce hack on [N.root.id].</span>"
+			N.root.execute("security -brute -[N.id]", H) // Trigger the root's security. Loudly.
+			connect(N.root, H)
+			return feed(H)
+		feedback = "<span class='warning'>Encryption Key Mismatch.</span>"
+		N.root.execute("security -alert -[N.id]", H) // Trigger the root's security. Quietly.
+		return feed(H)
+	for(var/datum/network/S in N.sub)
```

Comment:
Check https://github.com/tgstation/-tg-station/blob/master/CONTRIBUTING.md#dream-maker-quirkstricks, the jist of it is that when you provide a typepath in a for each Byond will automatically ```istype()``` the list data for you, even if it's not necessary, such as a situation like this where ```N.sub``` will only contain network datums, so why istype them all?
It's a micro optimisation but it helps also to highlight that the list's data is all the same (or subtype'd)